pennylane/ops/functions/equal.py

```diff
@@ -419,40 +419,69 @@ def _equal_pow(op1: Pow, op2: Pow, **kwargs):
     """Determine whether two Pow objects are equal"""
     if op1.z != op2.z:
         return False
```

Comment:
@albi3ro Should we be checking for trainability/interface of the exponents here? :thinking:

Reply:
Good catch! Potentially.  Most of the time we just pretend the exponent is a built-in, but we should probably start generalizing to any type of data.

Reply:
Can be done in a different PR, I guess? Or should it be included here?

Reply:
@dwierichs @albi3ro 
Thank you for your comment! 
I think those changes are also highly relevant to this Issue, so can I change them and include them in this PR?

I would like to confirm one point about the behavior of `_equal_exp`.
In `qml.exp`, is it necessary to perform comparisons that take into account numerical errors in the exponential part?

```
>>> op1 = qml.exp(qml.RX(1.2, wires=0), 2)
>>> op2 = qml.exp(qml.RX(1.2, wires=0), 2 + 1e-4)

# current behavior
>>> qml.equal(op1, op2, atol=1e-3)
False
# (because `op1.z != op2.z` return False)

# numerical error tolerant behavior
>>> qml.equal(op1, op2, atol=1e-3)
True
```

Reply:
@kenya-sk I was thinking about exactly that as well. We _could_ change this ,but might not want to, at this point. I'll leave this to @albi3ro to decide or carry to an internal discussion. Thanks for your thorough work on this!

Reply:
Sorry about not getting back to this earlier. 

At this point I'm just going to go ahead and merge this PR, but is definitely something we can do in the future.